packages/sanity/src/core/releases/tool/useBundlesMetadata.ts

```diff
@@ -0,0 +1,52 @@
+import {useEffect, useState} from 'react'
+
+import {useBundlesMetadataProvider} from '../contexts/BundlesMetadataProvider'
+
+export interface BundlesMetadata {
+  /**
+   * The number of documents with the bundle version as a prefix
+   */
+  documentCount: number
+  /**
+   * The last time a document in the bundle was edited
+   */
+  updatedAt: string | null
+}
+
+export const useBundlesMetadata = (bundleSlugs: string[]) => {
```

Comment:
This has been tested when called in multiple locations with different and overlapping bundle slugs, and all seems to work well. The current use is in only a single place ([`ReleasesOverview`](https://github.com/sanity-io/sanity/pull/7124/files#diff-5d02566c9762e8d3fc6511b5e2b518985b96f46fd0421ae80c8718d8823fe7f1), but in the future this will support batching all called to `useBundlesMetadata` and de-duping identical bundle slugs